action-scripts/merge_pr_jsons.py

```diff
@@ -23,43 +22,96 @@
 }
 ''')
 
+CHAIN_IDS = {
+    1: "mainnet",
+    3: "ropsten",
+    42161: "arbitrum",
+    100: "gnosis",
+    137: "polygon",
+    10: "optimism",
+}
+
+# Place your BIPs json into this directory under BIPs/<TARGET_DIR_WITH_BIPS>
+TARGET_DIR_WITH_BIPS = "00merging"
+
+
+def _parse_bip_json(file_path: str, chain: int) -> Optional[dict]:
+    """
+    In case file was created within given date bounds and for given chain -
+    parse it and return the data
+    """
+    # Check if the file is a json file
+    if not file_path.endswith(".json"):
+        return None
+    try:
+        with open(file_path, "r") as json_file:
+            data = json.load(json_file)
+            # Check if the file is a dictionary, not a list
+            if not isinstance(data, dict):
+                return None
+            # Check if chain id is the same as the one we are looking for
+            if int(data["chainId"]) == int(chain):
+                return data
+    except JSONDecodeError:
+        return None
+
+
 def main():
-    changed_files = sys.argv[1].split()
-    date = datetime.utcnow()
-    # Filter the list of added files for json files
-    json_files = [filename for filename in changed_files if filename.endswith(".json") and "BIPs/BIP" in filename]
-    txmap = {}
-    # Extract the relevant information from the JSON file
-    for json_file in json_files:
-        if debug:
-            print(f"Processing {json_file}")
-        # Get the JSON file from the repository
-        with open(json_file, "r") as json_data:
-            data = json.load(json_data)
-        # Extract the relevant information from the JSON file
-        chain_id = data["chainId"]
-        if chain_id not in txmap.keys():
-            txmap[chain_id]={}
-        safe = data["meta"]["createdFromSafeAddress"]
-        print(safe)
-        if safe not in txmap[chain_id].keys():
-            txmap[chain_id][safe] = []
-        txmap[chain_id][safe] = [*txmap[chain_id][safe], *data["transactions"]]
-        print(txmap)
+    current_week = datetime.utcnow().strftime("%U")
+    current_year = datetime.utcnow().year
+    # get root directory of the project:
+    root_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    bips_dir = os.path.join(root_dir, "BIPs")
+    dir_with_target_bips = os.path.join(bips_dir, TARGET_DIR_WITH_BIPS)
```

Comment:
BIPs should be read from a path provided as an input for now (via env var or argv)

BIPs should be written to 00merged/YYYY-#WW  (or 00batched, there's one example there)

Reply:
the destination directory for the given run won't exist when the report runs.

Reply:
That's clear. Gonna fix this

Reply:
Done